deny.toml

```diff
@@ -0,0 +1,30 @@
+# Project policies.
+#
+# These policies can be enforced using `cargo deny check`.
+
+[licenses]
+# Don't allow code with an unclear license.
+unlicensed = "deny"
+
+# Don't allow "copylefted" licenses unless they're listed below.
+copyleft = "deny"
+
+# Allow common non-restrictive licenses.
+allow = ["MIT", "Apache-2.0", "BSD-3-Clause", "BSD-2-Clause", "CC0-1.0"]
+
+# Many organizations ban AGPL-licensed code
+# https://opensource.google/docs/using/agpl-policy/
+deny = ["AGPL-3.0"]
+
+# Some slightly odder licenses. ISC is an unrestrictive license but unusual, and
+# MPL-2.0 is long and complicated "copyleft" license that's less strict than the
+# GPL. 
```

Comment:
MPL is roughly comparable to the LGPL except the license boundary is per-file rather than per-library. Source modifications trigger additional clauses to make changes available similar to GPL/LGPL. There is also some language WRT reciprocal patent licensing similar to LGPLv3.

IMO it's generally safe to use MPL/LGPL software in proprietary products as long as we're cool with releasing modified MPL/LGPL sources which I think we are. Neither license is "infectious" like the GPL/AGPL.

Reply:
Oh, we're completely happy releasing sources.

But the MPL is a partial copyleft license, even if only at a per-file level, and it imposes unusually complicated rules. Since we literally have one single MPLed dependency, and we're planning to remove it for technical reasons, I've configured the license checker to forbid new MPL-only dependencies.